### ISSUE TITLE
error behavior 

### ISSUE THREAD

**Issue by sckott** [ropensci/rebird#69]
noticed when fixing `spocc`'s ebird integration that error behavior could be better. 

when the API key is not found or not set, there's a nice error message about this. 👍 

But if the API key is bad/not known by eBird, the error behavior is pretty bad:

```r
library(rebird)

# works fine
ebirdregion(loc = 'US', species = 'btbwar') 

# unset api key
Sys.unsetenv('EBIRD_KEY')

# fails nicely
ebirdregion(loc = 'US', species = 'btbwar')
#> Error: You must provide an API key from eBird.
#>     You can pass it to the 'key' argument or store it as
#>     an environment variable called EBIRD_KEY in your .Renviron file.
#>     If you don't have a key, you can obtain one from:
#>     https://ebird.org/api/keygen.

# change API key to an obvisouly bad value
Sys.setenv(EBIRD_KEY = 'a')

# now fails badly
ebirdregion(loc = 'US', species = 'btbwar')
#> Error: parse error: premature EOF
#> 
#>                      (right here) ------^
```

The problem is the data from eBird is being processed before checking for any errors - here https://github.com/ropensci/rebird/blob/master/R/zzz.R#L37-L38  There i a proper error on the above request (403 http status code), so you should be able to catch that and then proceed accordingly with a nice error for the user.

**Comment by sckott** [comment]
Another issue I found: when you pass ebird a bad value you only get NA back. eBird does it appears give back useful messages about what went wrong at least in some cases, eg., 

```
curl -H 'X-eBirdApiToken: token' https://ebird.org/ws2.0/data/obs/4444/recent/shshaw

{"errors":[{"status":"500","code":"http.status.500","title":"Unable to find matching region type for 4444"}]}
```

passing 

> Unable to find matching region type for 4444

to the user would be really helpful to the user i reckon

**Comment by sebpardo** [comment]
Hi @sckott , thanks for pointing this out. As you suggest this should be easy to address. Adding `httr::stop_for_status(tt)` within `ebird_GET()` right at [this point](https://github.com/ropensci/rebird/blob/d0b9889d1114e4219506447bb10b25913321e172/R/zzz.R#L36) results in the following:

```r
Sys.unsetenv('EBIRD_KEY')
Sys.setenv(EBIRD_KEY = 'a')
ebirdregion(loc = 'US', species = 'btbwar')
#> Error in ebird_GET(url, args, key = key, ...) : Forbidden (HTTP 403).
```

Do you think just providing the error message from the GET request is enough?

**Comment by sckott** [comment]
> Do you think just providing the error message from the GET request is enough?

perhaps. i know what 403 status means, but i imagine most people don't 😸  - you could leave that as is, and just put in the docs what that error message means.  Some APIs return a response body with a message about auth on a 403, but not ebird

**Comment by sckott** [comment]
The response body

`{"errors":[{"status":"500","code":"http.status.500","title":"Unable to find matching region type for 4444"}]}`

should still be there in the response within the R session, its just that stop_for_status only gives the status code and the message associated with that code following https://en.wikipedia.org/wiki/List_of_HTTP_status_codes - so you could parse that response body yourself if it's there, and if not just fall back to what stop_for_status does

**Comment by sebpardo** [comment]
Closed by #70

**Comment by sebpardo** [comment]
@sckott let me know if you find any bad error messages.
